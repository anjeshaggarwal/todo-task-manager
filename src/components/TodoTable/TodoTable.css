#todo-table {
    display: inline-block;
    background-color: bisque;
    padding: 5px 20px 5px 5px;
    overflow: auto;
    max-height: calc(100vh - 100px);
}