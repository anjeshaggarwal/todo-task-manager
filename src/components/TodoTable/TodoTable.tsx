import React from 'react'
import TodoTask from '../TodoTask/TodoTask';
import './TodoTable.css'

interface IProps {
    tasks: any
    onTaskClick: (task: string) => void
}

const TodoTable = (props: IProps) => {
    return <div id='todo-table'>
        <ul data-testid="task-list">
            {Object.keys(props.tasks).map((task: string, index: number) => <TodoTask onClick={props.onTaskClick} key={`${task}`} text={task} status={props.tasks[task]} />)}
        </ul>
    </div>
}

export default TodoTable;