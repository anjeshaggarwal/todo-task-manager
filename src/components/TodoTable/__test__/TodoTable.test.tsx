import ReactDOM from 'react-dom'
import TodoTable from './../TodoTable'
import { render, cleanup, fireEvent } from '@testing-library/react';
import React from 'react';
import renderer from 'react-test-renderer'

afterEach(cleanup)

it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<TodoTable tasks={{}} onTaskClick={jest.fn} />, div)
})

it('renders correctly', () => {
    const { getByTestId } = render(<TodoTable tasks={{}} onTaskClick={jest.fn} />)
    expect(getByTestId('task-list')).toBeTruthy()
})

it('match snapshot', () => {
    const tree = renderer.create(<TodoTable tasks={{ Task1: false, Task2: true }} onTaskClick={jest.fn} />).toJSON()
    expect(tree).toMatchSnapshot()
})

it('have correct length', () => {
    const { getAllByTestId } = render(<TodoTable tasks={{ Task1: false, Task2: true }} onTaskClick={jest.fn} />)
    expect(getAllByTestId('list-item')).toHaveLength(2)
})

it('function called', () => {
    const taskClicked = jest.fn()
    const { getAllByTestId } = render(<TodoTable tasks={{ Task1: false, Task2: true }} onTaskClick={taskClicked} />)
    fireEvent.click(getAllByTestId('list-span')[0])
    expect(taskClicked).toHaveBeenCalled()
})