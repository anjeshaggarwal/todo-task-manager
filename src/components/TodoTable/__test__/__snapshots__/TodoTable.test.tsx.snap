// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot 1`] = `
<div
  id="todo-table"
>
  <ul
    data-testid="task-list"
  >
    <li
      className=""
      data-testid="list-item"
    >
      <span
        data-testid="list-span"
        onClick={[Function]}
      >
        Task1
      </span>
    </li>
    <li
      className="strike"
      data-testid="list-item"
    >
      <span
        data-testid="list-span"
        onClick={[Function]}
      >
        Task2
      </span>
    </li>
  </ul>
</div>
`;
